### ISSUE TITLE
[BUG] Phalcon 2.0.* - merging configs with Closures kabooms

### ISSUE THREAD

**Issue by temuri416** [phalcon/cphalcon#10775]
Hi,

In Phalcon 2.0 the following code throws "*Call to undefined method Closure::count()*" exception:

```php
<?php

ini_set('display_startup_errors',1);
ini_set('display_errors',1);
error_reporting(-1);

use Phalcon\Config;

$configWithCallback = new Config([
    'callback' => function() { return 'Hello World!'; }
]);

$configWithCallback2 = new Config([
    'callback' => function() { return 'Hello World 2.0!'; },
    'callback2' => function() { return 'ABCD 123'; }
]);

try {
    print 'Merging configWithCallback and configWithCallback2' . PHP_EOL;
    $configWithCallback->merge($configWithCallback2);
    print 'Merged configWithCallback and configWithCallback2' . PHP_EOL;
    var_dump($configWithCallback);
} catch(\Exception $e) {
    var_dump($e);
}
unset($result);
```

Thanks!

**Comment by temuri416** [comment]
@andresgutierrez 

Any thoughts on that? It prevents me from trying Phalcon 2 with my app - it explodes right out of the gate.

**Comment by andresgutierrez** [comment]
I'm taking a look

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch

**Comment by temuri416** [comment]
Thanks, Andres.
